### PR TITLE
Audit config section, populate advanced-config section

### PR DESCRIPTION
**Pull request by LexLuthr**
Fixes parts of https://github.com/filecoin-project/lotus-docs/issues/77

- [x] Empty config section

- [x] Check links

- [x] Check aliases are in place

- [x] Rename config section to Setup

- [x] Add pre-requisite page

- [x] Make changes in menu

- [x] Do final check for 404 or incorrect redirects

- [x] Populate logging page - Separate rust log file

### CODE REVIEW

File: content/en/storage-providers/get-started/overview/index.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Lotus Mining for _mainnet_ has stringent minimal **[hardware requirements]({{< r
 
 The following guides are essential starting points for those willing to launch a Lotus miner:
 
-- The miner installation is covered in the [Installation guide]({{< relref "prerequisites" >}}). Once the installation is complete, the Lotus node, Lotus miner, and Lotus worker applications should all be installed.
+- The miner installation is covered in the [Installation guide]({{< relref "../../../lotus/install/prerequisites" >}}). Once the installation is complete, the Lotus node, Lotus miner, and Lotus worker applications should all be installed.
```
Comment (rjan90) [inline content/en/storage-providers/get-started/overview/index.md:26]: I have started a audit/revision of the overview-page already where this bullet-point is removed, so maybe you could revert this change, so that branch does not need rebasing.

File: content/en/storage-providers/setup/setup.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+---
+title: "Setup"
```
Comment (rjan90) [inline content/en/storage-providers/setup/setup.md:2]: Maybe rename this to `Initialize` to keep things in-line with what was [outline here](https://github.com/filecoin-project/lotus-docs/issues/77#issuecomment-1091139591). Or what do you think?

Comment (LexLuthr) [reply content/en/storage-providers/setup/setup.md:2]: Earlier plan was to initialize the miner in that page and move start to operate section. But if you remember our discussion, we decided to leave it as setup since we will be keeping everything in the same page and moving pre-requisites to another. What do you think?

**Comment by LexLuthr** [comment]
@TippyFlitsUK @rjan90 @Reiers Can you please add information to the logging page?

I need help with Rust logs. It is marked as TODO.
I also need more examples of reading miner logs apart from the 2 already there.

File: content/en/storage-providers/setup/prerequisites.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    storage-providers:
+        parent: "storage-providers-setup"
+        identifier: "storage-providers-setup-prerequisites"
+aliases:
+    - /docs/storage-providers/setup/
+weight: 100
+toc: true
+---
+
+Mining will only work if you fully cover the [minimal hardware requirements]({{< relref "hardware-requirements" >}}) for the network in which you will mine. As the mining process is very demanding for the machines on several aspects and relies on precise configuration, we strongly recommend Linux systems administration experience before embarking.
```
Comment (TippyFlitsUK) [inline content/en/storage-providers/setup/prerequisites.md:16]:
```suggestion
Mining will only work if you fully comply with the [minimal hardware requirements]({{< relref "hardware-requirements" >}}) for the network in which you will mine. The mining process is very resource-intensive and is dependent on precise configuration. We strongly recommend Linux systems administration experience before embarking.
```

File: content/en/storage-providers/setup/prerequisites.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+{{< /alert >}}
+
+
+## Optional prerequisites
+
+These prerequisites are optional and can be used on case by case basis. Please make sure to understand the use case before performing these steps.
+
+### Install extra dependencies
+
+This step is only if you are running an Nvidia GPU and would prefer to use OpenCL instead of CUDA. Please note that Cuda is after than OpenCL for sealing and proving workload. Most Linux distributions contain this package in their package manager:
```
Comment (TippyFlitsUK) [inline content/en/storage-providers/setup/prerequisites.md:132]:
```suggestion
This step is only necessary if you are running an Nvidia GPU and would prefer to use OpenCL instead of CUDA. Please note that Cuda is recommended over OpenCL for sealing and proving workload. Most Linux distributions contain this package in their package manager:
```

File: content/en/storage-providers/setup/setup.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+weight: 110
+toc: true
+---
+
+## Checklist before launch
+
+To summarize, make sure that:
+
+- The _worker address_ has some funds so that the miner can be initialized.
+- The following environment variables have been defined and will be available for any Lotus miner runs:
```
Comment (TippyFlitsUK) [inline content/en/storage-providers/setup/setup.md:22]:
```suggestion
- The following environment variables have been defined and will be available for all Lotus miner instances:
```

File: content/en/storage-providers/setup/setup.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  export FIL_PROOFS_MAXIMIZE_CACHING=1
+  export FIL_PROOFS_USE_GPU_COLUMN_BUILDER=1 # When having GPU.
+  export FIL_PROOFS_USE_GPU_TREE_BUILDER=1   # When having GPU.
+  export FIL_PROOFS_PARAMETER_CACHE=/fast/disk/folder # > 100GiB!
+  export FIL_PROOFS_PARENT_CACHE=/fast/disk/folder2   # > 50GiB!
+  ```
+
+- Parameters have been prefetched to the cache folders specified above.
+- The system has enough swap and it is active.
+- The lotus api has been configured and variable has been exported in the environment where lotus miner runs.
```
Comment (TippyFlitsUK) [inline content/en/storage-providers/setup/setup.md:37]:
```suggestion
- The lotus api has been configured and the variable has been exported in the environment where lotus miner runs.
```

File: content/en/storage-providers/setup/setup.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+## Miner initialization
+
+Before starting your miner for the first time run:
+
+```sh
+lotus-miner init --owner=<address>  --worker=<address> --no-local-storage
+```
+
+- The `--no-local-storage` flag is used so that we can later configure [specific locations for storage]({{< relref "custom-storage-layout" >}}). This is optional but recommended.
+- The Lotus Miner configuration folder is created at `~/.lotusminer/` or `$LOTUS_MINER_PATH` if set.
```
Comment (TippyFlitsUK) [inline content/en/storage-providers/setup/setup.md:49]:
```suggestion
- The Lotus Miner configuration folder is created in `~/.lotusminer/` or `$LOTUS_MINER_PATH` if set.
```

File: content/en/storage-providers/setup/setup.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+systemctl start lotus-miner
+```
+
+{{< alert icon="warning" >}}
+**Do not proceed** from here until you have verified that your miner not only is running, but also [reachable on its public IP address]({{< relref "connectivity" >}}).
+{{< /alert >}}
+
+## Publishing the miner addresses
+
+Once the miner is up and running, publish your miner address (which you configured above) on the chain so that other nodes can talk to it directly and make deals:
```
Comment (TippyFlitsUK) [inline content/en/storage-providers/setup/setup.md:86]:
```suggestion
Once the miner is up and running, publish your miner address (which you configured above) to the chain so that other nodes can talk to it directly and make deals:
```

File: content/en/storage-providers/setup/setup.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+lotus-miner actor set-addrs /ip4/<YOUR_PUBLIC_IP_ADDRESS>/tcp/24001
+```
+
+## Next steps
+
+Your miner should now be preliminarily set up and running, but **there are still a few more recommended tasks** to be ready for prime-time:
+
+- Set up your [custom storage layout]({{< relref "custom-storage-layout" >}}) (required if you used `--no-local-storage`).
+- Edit the miner [configuration settings]({{< relref "configuration" >}}) to fit your requirements.
+- Learn what is a right moment to [shut down/restart your miner]({{< relref "maintenance" >}})
```
Comment (TippyFlitsUK) [inline content/en/storage-providers/setup/setup.md:98]:
```suggestion
- Learn when to safely [shut down/restart your miner]({{< relref "maintenance" >}})
```

**Review by TippyFlitsUK — COMMENTED** [review]
Very nice work Mayank!! Just waiting on confirmation of the RUST logging steps (testing now).

File: content/en/storage-providers/setup/setup.md (hunk 2 above)
Comment (rjan90) [reply content/en/storage-providers/setup/setup.md:2]: I agree about the content in the page. Was mostly about the wording, and having a `setup`-page within the `setup` section that seemed a bit repetitive/wrong. Maybe some English native speakers have some input here.

File: content/en/kb/add-swap/index.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+pinned: false
+types: ["article"]
+areas: ["swap", "Linux"]
+---
+
+## Add swap on lotus miner or worker node:
+
+If you only have 128 GiB of RAM, you will need to make sure your system provides at least an extra 256 GiB of very fast swap (preferably NVMe SSD) or you will be unable to seal sectors:
+
+```sh
```
Comment (johnnymatthews) [inline content/en/kb/add-swap/index.md:20]:
````suggestion
```shell
````

This is a tiny nit-pick, but use `shell` instead of `sh`.

File: content/en/kb/failed-to-create-datastore/index.md [hunk 11]
```diff
@@ -13,6 +13,7 @@ types: ["solution"]
 areas: ["sealing", "Lotus Worker", "Lotus Miner"]
 ---
 
+## Problem
```
Comment (johnnymatthews) [inline content/en/kb/failed-to-create-datastore/index.md:16]:
```suggestion
## Problem

```

File: content/en/storage-providers/advanced-configurations/logging.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+menu:
+    storage-providers:
+        parent: "storage-providers-advanced-configurations"
+weight: 160
+toc: true
+---
+
+## Log file configuration
+
+The Lotus miner process generates two types of logs, namely, Go logs and Rust logs. Each of these logs can redirecting to individual files.
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/logging.md:15]:
```suggestion
The `lotus-miner` process generates Go logs and Rust logs. Each of these logs can redirecting to individual files.
```

When talking about the Lotus miner application, use code blocks: `lotus-miner`.

Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/logging.md:15]: > Each of these logs can redirecting to individual files.

I don't understand this sentence.

File: content/en/storage-providers/advanced-configurations/logging.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+{{< alert icon="callout" >}}
+This section is not applicable if you are running lotus-miner as a systemd service.
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/logging.md:18]:
```suggestion
This section is not applicable if you are running `lotus-miner` as a systemd service.
```

File: content/en/storage-providers/advanced-configurations/logging.md [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+{{< /alert >}}
+
+### Redirect GO logs to a file
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/logging.md:21]:
```suggestion
### Redirect Go logs to a file
```

File: content/en/storage-providers/advanced-configurations/logging.md [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+By default, lotus-miner redirect all logs to the standard output if not running as a systemd service. To change this behavior, add the following variable to the `.bashrc` file and restart the Lotus miner process to start redirecting all logs to the file.
+
+```shell
+export GOLOG_OUTPUT=FILE >> ~/.bashrc
+export GOLOG_FILE="$HOME/miner.log" >> ~/.bashrc && source ~/.bashrc
+```
+
+
+### Redirect Rust logs to a standard output
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/logging.md:31]:
```suggestion
### Redirect Rust logs to a standard output

```

File: content/en/storage-providers/advanced-configurations/logging.md [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+By default the `fil_logger` library used by `rust-fil-proof` doesn't log anything. You can change this by setting the RUST_LOG environment variable to another level. This will show log output on stderr. It can help debug error during the sealing and windowPost.
+
+```shell
+export RUST_LOG=info >> ~/.bashrc && source ~/.bashrc
+```
+
+The log-level can be chose between 5 options
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/logging.md:38]:
```suggestion
The log-level can be chose between 5 options:

```

File: content/en/storage-providers/advanced-configurations/logging.md [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+
+## Change logging verbosity
+
+The verbosity of the `lotus` and `lotus-miner` logs can be changed without restarting the lotus. The following command can be used to list different subsystems within the `lotus-miner` process and change the verbosity of individual subsystem to get more/less detailed logs.
+
+```shell
+lotus-miner log list
+```
+
+To change the verbosity, please run
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/logging.md:53]:
```suggestion
To change the verbosity, please run:
```

File: content/en/storage-providers/advanced-configurations/logging.md [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+
+## Searching through logs
+
+1. To look for logs related to mining a block, you can check with following commands.
+    ```shell
+    cat /path/to/log | grep "mined"
+
+2. To look for logs related to winning a block, you can check with following commands.
+    ```shell
+    cat /path/to/log | grep "isEligible"
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/logging.md:79]: The formatting here is broken. Are we missing some steps?

File: content/en/storage-providers/advanced-configurations/market.md [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+For example, if `PublishMsgPeriod` is 1 hour:
+
+- At 1:00 pm, deal 1 is ready to publish. Lotus will wait until 2:00 pm for other deals to be ready before sending `PublishStorageDeals`.
+- At 1:30 pm, Deal 2 is ready to publish
+- At 1:45 pm, Deal 3 is ready to publish
+- At 2:00pm, lotus publishes Deals 1, 2, and 3 in a single `PublishStorageDeals` message.
+
+If `MaxDealsPerPublishMsg` is 2, then in the above example, when deal 2 is ready to be published at 1:30, Lotus would immediately publish Deals 1 & 2 in a single `PublishStorageDeals` message. Deal 3 would be published in a subsequent `PublishStorageDeals` message.
+
+> If any of the deals in the `PublishStorageDeals` fails validation upon execution, or if the start epoch has passed, all deals will fail to be published.
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/market.md:86]:
```suggestion
> If any of the deals in the `PublishStorageDeals` fails validation upon execution, or if the start epoch has passed, all deals will fail to be published.
```

If you need to add a call out use theme alerts:

```plaintext
{{< alert icon="👉" >}}
_Markdown_ and <em>HTML</em> will be rendered.
{{< /alert >}}
```

Do not use quote blocks for call outs. Quote blocks are for quotes.

File: content/en/storage-providers/advanced-configurations/sealing.md [hunk 20]
```diff
@@ -0,0 +1,244 @@
...
+- `MaxPreCommitBatch` is the maximum amount of sectors' pre-commitments to batch in one `PreCommitSectorsBatch` message. According to FIP-0008, this values is up to 256.
+- `PreCommitBatchWait` is how long to wait before submitting the current batch. Note: the ticket of pre-commitment has an expiration of approximately **31.5 hours**, one sector's pre-commit ticket expires **WILL** cause the whole message to fail. Therefore, **we recommend miners to set this value lower than 30 hours.**
+- `PreCommitBatchSlack` is the time buffer to forcefully submit the current batch before any of the sector's pre-commit ticket or a deal will expire. For example, if this value is set to 1 hour, which is 120 epochs, then a `PreCommitSectorsBatch` message will be submitted for the existing batch 120 epochs before the earliest epoch among precommits' tickets and deal's start epochs in this batch. **We recommend you to set a longer slack to prevent message failures due to expirations.**
+
+> Note, the current batch will be sent if any of `MaxPreCommitBatch`, `PreCommitBatchWait` or `PreCommitBatchSlack` is hit.
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/sealing.md:87]: If you need to add a call out use theme alerts:

```plaintext
{{< alert icon="👉" >}}
_Markdown_ and <em>HTML</em> will be rendered.
{{< /alert >}}
```

Do not use quote blocks for call outs. Quote blocks are for quotes.

File: content/en/storage-providers/advanced-configurations/sealing.md [hunk 21]
```diff
@@ -0,0 +1,244 @@
...
+      Base = "0 FIL"
+      PerSector = "0.03 FIL"
+
+```
+
+Depending on the network congestion, the base fee for a message may grow or decrease. Your gas limits will have to be larger than the base fee for the messages to be included. A very large max fee can, however, result in the quick burning of funds when the base fee is very high, as the miner automatically submits messages during normal operation, so be careful about this. It is also necessary to have more funds available than any max fee set, even if the actual fee will be far less than the max fee set.
+
+Set the maximum cost you are willing to pay for onboarding **per** sector in `MaxPreCommitBatchGasFee.PerSector`/`MaxCommitBatchGasFee.PerSector` to avoid unexpected high costs.
+
+> Note: The current `MaxCommitBatchGasFee.PerSector` is enough to aggregate proofs for 6 sectors. Adjust this respectively according to your operation. **If the value is too low, the message may wait in the mempool for a long while. If you don't have enough funds, the message will not be sent.**
```
Comment (johnnymatthews) [inline content/en/storage-providers/advanced-configurations/sealing.md:244]: If you need to add a call out use theme alerts:

```plaintext
{{< alert icon="👉" >}}
_Markdown_ and <em>HTML</em> will be rendered.
{{< /alert >}}
```

Do not use quote blocks for call outs. Quote blocks are for quotes.

File: content/en/storage-providers/setup/prerequisites.md [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+
+This step is only necessary if you are running an Nvidia GPU and would prefer to use OpenCL instead of CUDA. Please note that Cuda is recommended over OpenCL for sealing and proving workload. Most Linux distributions contain this package in their package manager:
+
+```shell
+sudo apt update -y && sudo apt install -y nvidia-opencl-icd -y
+```
+
+### Optional Performance tweaks
+
+```sh
```
Comment (johnnymatthews) [inline content/en/storage-providers/setup/prerequisites.md:140]:
```suggestion
```shell
```

File: content/en/storage-providers/setup/prerequisites.md [hunk 23]
```diff
@@ -0,0 +1,157 @@
...
+```
+
+The `BELLMAN_CPU_UTILIZATION` is an optional variable to designate a proportion of the multi-exponentiation calculation to be moved to a CPU in parallel to the GPU. This is an effort to keep all the hardware occupied. The interval must be a number between `0` and `1`. The value `0.875` is a good starting point, but you should experiment with it if you want an optimal setting. Different hardware setups wil
+l result in different values being optimal. Omitting this environment variable might also be optimal.
+
+### Downloading parameters
+
+To download the parameters do:
+
+```sh
```
Comment (johnnymatthews) [inline content/en/storage-providers/setup/prerequisites.md:152]:
```suggestion
```shell
```

File: content/en/storage-providers/setup/setup.md [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+---
+
+## Checklist before launch
+
+To summarize, make sure that:
+
+- The _worker address_ has some funds so that the miner can be initialized.
+- The following environment variables have been defined and will be available for all Lotus miner instances:
+
+  ```
```
Comment (johnnymatthews) [inline content/en/storage-providers/setup/setup.md:24]:
```suggestion
  ```shell
```

File: content/en/storage-providers/setup/setup.md [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+- Parameters have been prefetched to the cache folders specified above.
+- The system has enough swap and it is active.
+- The lotus api has been configured and the variable has been exported in the environment where lotus miner runs.
+- Wallets have been created for the owner and worker address.
+
+## Miner initialization
+
+Before starting your miner for the first time run:
+
+```sh
```
Comment (johnnymatthews) [inline content/en/storage-providers/setup/setup.md:44]:
```suggestion
```shell
```

File: content/en/storage-providers/setup/setup.md [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+...
+```
+
+Once you start your miner, [make sure you can connect to its public IP/port]({{< relref "connectivity" >}}).
+
+## Starting the miner
+
+You are now ready to start your Lotus miner:
+
+```sh
```
Comment (johnnymatthews) [inline content/en/storage-providers/setup/setup.md:70]:
```suggestion
```shell
```

File: content/en/storage-providers/setup/setup.md [hunk 27]
```diff
@@ -0,0 +1,102 @@
...
+lotus-miner run
+```
+
+or if you are using the systemd service file:
+
+```sh
```
Comment (johnnymatthews) [inline content/en/storage-providers/setup/setup.md:76]:
```suggestion
```shell
```

File: content/en/storage-providers/setup/setup.md [hunk 28]
```diff
@@ -0,0 +1,102 @@
...
+
+{{< alert icon="warning" >}}
+**Do not proceed** from here until you have verified that your miner not only is running, but also [reachable on its public IP address]({{< relref "connectivity" >}}).
+{{< /alert >}}
+
+## Publishing the miner addresses
+
+Once the miner is up and running, publish your miner address (which you configured above) to the chain so that other nodes can talk to it directly and make deals:
+
+```sh
```
Comment (johnnymatthews) [inline content/en/storage-providers/setup/setup.md:88]:
```suggestion
```shell
```

**Review by johnnymatthews — CHANGES_REQUESTED** [review]
Changes needed:

- When mentioning the Lotus miner application, always surround it in quotes and add a dash. So `lotus-miner`, not _Lotus miner_.
- Don't use quote blocks for call outs.
- If files have moved around you **need** to add redirects/aliases.

There are **a lot** of changes in this PR. Next time, split your changes between multiple smaller PRs, it makes things much easier to review and figure out what's going wrong when something breaks.

File: content/en/storage-providers/operate/custom-storage-layout.md [hunk 29]
```diff
@@ -45,6 +45,6 @@ lotus-miner storage list
 
 ## Updating locations
 
-To move data from one location to another, follow the instructions in the [Miner lifecycle guide]({{< relref "lifecycle#changing-storage-locations" >}}).
+To move data from one location to another, follow the instructions in the [Miner maintenance guide]({{< relref "maintenance#changing-storage-locations" >}}).
```
[the remark this answers is not in the document]
Comment (LexLuthr) [reply content/en/storage-providers/operate/custom-storage-layout.md:48]: Storage provider maintenance would imply an overall maintenance. This guide just deals with lotus-miner process maintenance. Does the name make sense in that case?

**Review by johnnymatthews — APPROVED** [review]
Thanks for those changes!
